src/renderer/viz/expressions/ramp.js

```diff
@@ -287,28 +284,31 @@ export default class Ramp extends BaseExpression {
     _computeGLTextureIfNeeded (gl) {
         const texturePixels = this._computeTextureIfNeeded();
 
-        if (this._GLtexCategories !== this.input.numCategories) {
+        if (this._GLtexCategories !== this.input.numCategories || this.isAnimated()) {
```

Comment:
Uploading the texture is performance intensive and I think this is only needed (regarding animation) when the palette is animated, not when the rest of the ramp parameters are animated.